Automattic/pocket-casts-android
# Add remote feature flags provider (in release build only)

## Description

This adds Firebase remote feature flags provider (in release build only). Values from this provider override those of the default provider. A feature should be enabled in this provider only when it has to be controlled through `Firebase` `RemoteConfig`.

## Testing Instructions

Added `FEATURE_FLAG_TEST` in `Firebase` `RemoteConfig` with the value set to true for the release build. 

The following test checks that the value from `FirebaseRemoteFeatureFlagProvider` overrides that of the default provider for the release build (`StoreFeatureFlagProvider`) where it is set to false.

1. Apply [this patch](https://github.com/Automattic/pocket-casts-android/files/11757161/Test_firebase_remote_feature_flag.patch)
2. Fresh install `release` build
3. Go to `About` page
4. Notice in logs "test feature flag enabled"

## Checklist
 
#### I have tested any UI changes...
